engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -115,15 +114,18 @@ class WorkflowManagerActor(config: Config,
   }
 
   private def addShutdownHook() = {
-    // Only abort jobs on SIGINT if the config explicitly sets system.abortJobsOnTerminate = true.
+    // Only abort jobs on SIGINT if the config explicitly sets system.abort-jobs-on-terminate = true.
     val abortJobsOnTerminate =
       config.getConfig("system").as[Option[Boolean]]("abort-jobs-on-terminate").getOrElse(false)
 
     if (abortJobsOnTerminate) {
       sys.addShutdownHook {
-        logger.info(s"$tag: Received shutdown signal. Aborting all running workflows...")
+        logger.info(s"$tag: Received shutdown signal.")
         self ! AbortAllWorkflowsCommand
-        Await.result(donePromise.future, Duration.Inf)
+        while (stateData != null && stateData.workflows.nonEmpty) {
+          log.info(s"Waiting for ${stateData.workflows.size} workflows to abort...")
+          Thread.sleep(1000)
```

Comment:
ToL: Someday... I'd like to see us wait for "abort handled" messages to come back instead of sleeping.